core/lib/spree/core/product_duplicator.rb

```diff
@@ -70,5 +70,12 @@ def reset_properties
         end
       end
     end
+
+    def generate_new_sku(sku)
+      loop do
+        sku = "COPY OF #{sku}"
+        break sku unless Spree::Variant.where(sku: sku).first
```

Comment:
Agree, `.where`...`.first` is bad https://github.com/bbatsov/rails-style-guide#find_by